File: app/src/main/java/com/krepchenko/besafe/ui/activities/PinActivity.java
package com.krepchenko.besafe.ui.activities;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.KeyEvent;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;

import com.krepchenko.besafe.R;
import com.krepchenko.besafe.core.SafeApplication;
import com.krepchenko.besafe.crypt.CryptManager;

/**
 * Created by Ann on 19.10.2015.
 */
public class PinActivity extends AppCompatActivity implements TextView.OnEditorActionListener, TextWatcher {

    private EditText etPass;
    private SafeApplication application;

    public static void launch(Activity activity) {
        Intent intent = new Intent(activity, PinActivity.class);
        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
        activity.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        application = (SafeApplication) getApplicationContext();
        setContentView(R.layout.activity_pin);
        initViews();
    }

    private void initViews() {
        etPass = (EditText) findViewById(R.id.pin_pass_et);
        etPass.setOnEditorActionListener(this);
        etPass.addTextChangedListener(this);
        if (!application.getSharedManager().isHintViewed()) {
            findViewById(R.id.pin_user_hint).setVisibility(View.VISIBLE);
        }
    }

    @Override
    public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
        if (actionId == EditorInfo.IME_ACTION_DONE || v.getText().length() == 4) {
            if (v.getId() == R.id.pin_pass_et) {
                if (etPass.getText().length() != 4) {
                    etPass.setError(getString(R.string.error_pin));
                } else {
                    submitText();
                }
            }
        }
        return false;
    }

    private void submitText(){
        ((SafeApplication) getApplicationContext()).getSharedManager().setHintViewed();
        MainActivity.launch(this, CryptManager.getEncryptedPass(etPass.getText().toString()));
    }

    @Override
    public void beforeTextChanged(CharSequence s, int start, int count, int after) {

    }

    @Override
    public void onTextChanged(CharSequence s, int start, int before, int count) {
        if (s.length() == 4) {
            ((SafeApplication) getApplicationContext()).getSharedManager().setHintViewed();
            MainActivity.launch(this, CryptManager.getEncryptedPass(etPass.getText().toString()));
        }
    }

    @Override
    public void afterTextChanged(Editable s) {

    }
}

File: app/src/main/java/com/krepchenko/besafe/ui/activities/PinActivity.kt
package com.krepchenko.besafe.ui.activities

import android.app.Activity
import android.app.LoaderManager
import android.content.Intent
import android.content.Loader
import android.database.Cursor
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.Editable
import android.text.TextWatcher
import android.view.KeyEvent
import android.view.View
import android.view.inputmethod.EditorInfo
import android.widget.EditText
import android.widget.TextView
import com.krepchenko.besafe.R
import com.krepchenko.besafe.core.SafeApplication
import com.krepchenko.besafe.crypt.CryptManager
import com.krepchenko.besafe.db.SafeEntity

/**
 * Created by Ann on 19.10.2015.
 */
class PinActivity : AppCompatActivity(), TextView.OnEditorActionListener, TextWatcher, LoaderManager.LoaderCallbacks<Cursor> {




    private var etPass: EditText? = null
    private var application: SafeApplication? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        application = applicationContext as SafeApplication
        setContentView(R.layout.activity_pin)
        initViews()
    }

    private fun initViews() {
        etPass = findViewById(R.id.pin_pass_et)
        etPass!!.setOnEditorActionListener(this)
        etPass!!.addTextChangedListener(this)
        if (!application!!.sharedManager.isHintViewed) {
            findViewById<View>(R.id.pin_user_hint).visibility = View.VISIBLE
        }
    }

    override fun onEditorAction(v: TextView, actionId: Int, event: KeyEvent): Boolean {
        if (actionId == EditorInfo.IME_ACTION_DONE || v.text.length == 4) {
            if (v.id == R.id.pin_pass_et) {
                if (etPass!!.text.length != 4) {
                    etPass!!.error = getString(R.string.error_pin)
                } else {
                    submitText()
                }
            }
        }
        return false
    }

    private fun submitText() {
        (applicationContext as SafeApplication).sharedManager.setHintViewed()

        dbSetup()

        MainKeyActivity.launch(this, CryptManager.getEncryptedPass(etPass!!.text.toString()))
    }

    private fun dbSetup() {


    }

    override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {

    }

    override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
        if (s.length == 4) {
            (applicationContext as SafeApplication).sharedManager.setHintViewed()
            MainKeyActivity.launch(this, CryptManager.getEncryptedPass(etPass!!.text.toString()))
        }
    }

    override fun afterTextChanged(s: Editable) {

    }

    companion object {

        fun launch(activity: Activity) {
            val intent = Intent(activity, PinActivity::class.java)
            intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK)
            activity.startActivity(intent)
        }
    }


    @Override
    public Loader<Cursor> onCreateLoader(int id, Bundle args)
    {

    }

    @Override
    public void onLoadFinished(Loader<Cursor> loader, Cursor data )
    {
        adapter.changeCursor(data);
    }

    @Override
    public void onLoaderReset(Loader<Cursor> loader)
    {
        adapter.changeCursor(null);
    }
}
